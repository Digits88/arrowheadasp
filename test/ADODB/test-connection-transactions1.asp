<!--#include file="../config/ADODB.inc"-->
Here is a test of transactions.<br>
<%
Dim CN
Sub List_Table_Contents(CN)
    Set RS = CN.Execute("SELECT StringField, IntField FROM Test ORDER BY IntField")
    while not RS.EOF
        %>StringField: "<%=RS("StringField")%>", IntField: "<%=RS("IntField")%>"<br>
<%
        RS.MoveNext
    wend
End Sub
%>

Testing transactions.

Opening DB...
<%
Set CN = OpenConnection()
%>

List what's in the table initially...
<%
Call List_Table_Contents(CN)
%>

Begin a transaction...<br>
TransID: <%= CN.BeginTrans() %>

Add some elements to table...

<%
CN.Execute("INSERT INTO Test(StringField, IntField) VALUES ('A', 101);")
CN.Execute("INSERT INTO Test(StringField, IntField) VALUES ('B', 102);")
%>

List what's in the table now...
<%
Call List_Table_Contents(CN)
%>

Rollback...

<%
CN.RollbackTrans()
%>

List what's in the table after rollback...
<%
Call List_Table_Contents(CN)
%>

Close the database...
<%
CN.Close()
%>

