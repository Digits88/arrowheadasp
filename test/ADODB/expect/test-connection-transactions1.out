
Here is a test of transactions.<br>


Testing transactions.

Opening DB...


List what's in the table initially...
StringField: "First Row", IntField: "1"<br>
StringField: "Second Row", IntField: "2"<br>
StringField: "Third Row", IntField: "3"<br>
StringField: "Fourth Row", IntField: "4"<br>
StringField: "Fifth Row", IntField: "5"<br>
StringField: "Sixth Row", IntField: "6"<br>
StringField: "Seventh Row", IntField: "7"<br>
StringField: "Eigth Row", IntField: "8"<br>
StringField: "Ninth Row", IntField: "9"<br>
StringField: "Tenth Row", IntField: "10"<br>
StringField: "Eleventh Row", IntField: "11"<br>
StringField: "Twelfth Row", IntField: "12"<br>
StringField: "Thirteenth Row", IntField: "13"<br>
StringField: "Fourteenth Row", IntField: "14"<br>
StringField: "Fifteenth Row", IntField: "15"<br>


Begin a transaction...<br>
TransID: 1

Add some elements to table...



List what's in the table now...
StringField: "First Row", IntField: "1"<br>
StringField: "Second Row", IntField: "2"<br>
StringField: "Third Row", IntField: "3"<br>
StringField: "Fourth Row", IntField: "4"<br>
StringField: "Fifth Row", IntField: "5"<br>
StringField: "Sixth Row", IntField: "6"<br>
StringField: "Seventh Row", IntField: "7"<br>
StringField: "Eigth Row", IntField: "8"<br>
StringField: "Ninth Row", IntField: "9"<br>
StringField: "Tenth Row", IntField: "10"<br>
StringField: "Eleventh Row", IntField: "11"<br>
StringField: "Twelfth Row", IntField: "12"<br>
StringField: "Thirteenth Row", IntField: "13"<br>
StringField: "Fourteenth Row", IntField: "14"<br>
StringField: "Fifteenth Row", IntField: "15"<br>
StringField: "A", IntField: "101"<br>
StringField: "B", IntField: "102"<br>


Rollback...



List what's in the table after rollback...
StringField: "First Row", IntField: "1"<br>
StringField: "Second Row", IntField: "2"<br>
StringField: "Third Row", IntField: "3"<br>
StringField: "Fourth Row", IntField: "4"<br>
StringField: "Fifth Row", IntField: "5"<br>
StringField: "Sixth Row", IntField: "6"<br>
StringField: "Seventh Row", IntField: "7"<br>
StringField: "Eigth Row", IntField: "8"<br>
StringField: "Ninth Row", IntField: "9"<br>
StringField: "Tenth Row", IntField: "10"<br>
StringField: "Eleventh Row", IntField: "11"<br>
StringField: "Twelfth Row", IntField: "12"<br>
StringField: "Thirteenth Row", IntField: "13"<br>
StringField: "Fourteenth Row", IntField: "14"<br>
StringField: "Fifteenth Row", IntField: "15"<br>


Close the database...
